pkg/ocibundle/tools/overlay_linux.go

```diff
@@ -84,20 +102,41 @@ func CreateOverlayTmpfs(bundlePath string, sizeMiB int) error {
 }
 
 func prepareOverlay(bundlePath, overlayDir string) error {
+	var err error
+
 	upperDir := filepath.Join(overlayDir, "upper")
-	if err := os.Mkdir(upperDir, 0o755); err != nil {
-		return fmt.Errorf("failed to create %s: %s", upperDir, err)
+	_, err = os.Stat(upperDir)
+	if err != nil {
```

Comment:
Not sure I understand your intent on this one. Unless I'm missing something, rewriting it with this pattern will have to involve consecutive `if` statements that look like this:

```
if err != nil && !os.IsNotExist(err) {
  return err
}

if err != nil {
	err := os.Mkdir(upperDir, 0o755)
	if err != nil {
		return fmt.Errorf("failed to create %s: %s", upperDir, err)
	}
}
```

(Since, if the error occurred and it _was_ a not-exist scenario, we attempt to create the directory.)

Is that what you had in mind?

NB: I don't think we should invert the order of the two if-blocks here, because it'll create confusion about which `err` value we're looking at at a given point, due to the error checking for `os.Mkdir`. (Technically, `err` gets its own `:=` inside the conditional, so it won't overwrite the outer `err`, but writing code that only works by virtue of that kind of scope subtlety seems bad from a readability perspective, as does starting to mess with `err1` and `err2` variables.)